src/analysis/retail/monk/shared/hero/MasterOfHarmony/talents/Coalesence.tsx

```diff
@@ -0,0 +1,89 @@
+import { CAST_BUFFER_MS } from 'analysis/retail/monk/mistweaver/normalizers/EventLinks/EventLinkConstants';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateEffectiveDamage, calculateEffectiveHealing } from 'parser/core/EventCalculateLib';
+import Events, { DamageEvent, HealEvent } from 'parser/core/Events';
+import Combatants from 'parser/shared/modules/Combatants';
+import { COALESENCE_INCREASE } from '../constants';
+import Enemies from 'parser/shared/modules/Enemies';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+
+class Coalesence extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+  protected combatants!: Combatants;
+  protected enemies!: Enemies;
+  totalDmg: number = 0;
+  totalHealing: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.COALESCENCE_TALENT);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.onHeal);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+  }
+
+  private onHeal(event: HealEvent) {
+    const target = this.combatants.getEntity(event);
+    if (
```

Comment:
We can't test yet but i'm going to assume aspect of harmony wont increase itself so it'll have to be filtered out

Reply:
Oh yea probably not, will put the changes in but test on tuesday before committing.